### PR TITLE
[QF-3120] End of surah section - explore card

### PR DESCRIPTION
**Pull request by yousefdergham**
## Summary

Adds the ExploreCard component to the End of Surah Section, providing users with quick access to
explore related content including Tafsir, Reflections, Lessons, and Q&A. 
The card displays
personalized suggestions from the chapter metadata API and includes a "My Quran" link for easy
access to user profiles.

This PR builds upon the ReadMoreCard functionality by adding a second card that enhances user
engagement and provides multiple pathways for deeper exploration of Quranic content.

Closes: [QF-3120](https://quranfoundation.atlassian.net/browse/QF-3120)

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as
      expected)
- [ ] 📝 Documentation update
- [ ] ♻️ Refactoring (no functional changes)

## Test Plan

- [x] Unit tests added/updated
- [ ] Integration tests added/updated
- [x] Manual testing performed

#### Scenario 1: Verify Explore Card Exists and Renders Correctly

**Objective**: Verify ExploreCard component renders with all expected elements

**Steps**:

1. Navigate to the end of a chapter (e.g., `/2` for Al-Baqarah)
2. Scroll to the bottom of the page
3. Verify the "End of Chapter" section appears
4. Verify two cards are displayed: Read More Card and Explore Card
5. Verify Explore Card header shows "Explore" title
6. Verify "My Quran" link is visible in the header with bookmark icon
7. Verify action buttons row displays below the header (Tafsir, Reflections, Lessons)
8. Verify suggestions section appears below buttons (if API returns suggestions)

---

#### Scenario 2: Action Buttons Functionality and Clickability

**Objective**: Verify all action buttons work correctly and open appropriate modals

**Steps**:

1. Navigate to end of chapter `/1` (Al-Fatiha)
2. Locate Explore Card action buttons
3. Click "Tafsir" button
   - Verify Tafsir modal opens
   - Verify URL updates correctly
   - Verify modal displays tafsir content
   - Close modal
4. Click "Reflections" button
   - Verify Reflections modal opens
   - Verify navigation URL is correct
   - Close modal
5. Click "Lessons" button
   - Verify Lessons modal opens
   - Verify correct verse is loaded
   - Close modal
6. Click "My Quran" link in header
   - Verify navigation to profile page (`/profile`)
   - Verify analytics event is logged

---

#### Scenario 3: Accessibility Testing

**Steps**:

1. **Keyboard Navigation**:

   - Tab through all interactive elements in Explore Card
   - Verify focus indicators are visible on all buttons and links
   - Verify Enter/Space activate buttons when focused
   - Verify tab order is logical (header → buttons → suggestions)

---

#### Scenario 4: Answers Button Visibility 

**Objective**: Verify Q&A (Answers) button only appears and is clickable when questions are
available

**Test Case 4A: Chapter WITH Answers**

1. Navigate to end of a chapter known to have Q&A (e.g., `/2` - Al-Baqarah)
2. Verify "Answers" button is **visible** in Explore Card action buttons row
3. Verify button displays correct icon and label
4. Click "Answers" button
   - Verify button is **clickable** and responsive
   - Verify Questions modal opens
   - Verify answers are displayed in modal
   - Verify analytics event `end_of_surah_answers` is logged
5. Close modal

**Test Case 4B: Chapter WITHOUT Answers**

1. Navigate to end of a chapter without Q&A (e.g., `/114/6` - An-Nas)
2. Verify "Answers" button is **NOT visible** in Explore Card
3. Verify other buttons (Tafsir, Reflections, Lessons) still render correctly
4. Verify no layout shift or spacing issues when Answers button is hidden
5. Verify card layout remains correct

---

### Code Quality

- [x] I have performed a **self-review** of my code (file by file)
- [x] My code follows the [project style guidelines](/.github/copilot-instructions.md)
- [x] No `any` types used (or justified if unavoidable)
- [x] No unused code, imports, or dead code included
- [x] Complex logic has inline comments explaining "why"
- [x] Functions are under 30 lines and follow single responsibility

### Testing & Validation

- [x] All tests pass locally (`yarn test`)
- [x] Linting passes (`yarn lint`)
- [ ] Build succeeds (`yarn build`)
- [x] Edge cases and error scenarios are handled

### Documentation

- [x] Code is self-documenting with clear naming
- [ ] README updated (if adding features or setup changes)
- [x] Inline comments added for complex logic

### Localization (if UI changes)

- [x] All user-facing text uses `next-translate`
- [x] Only English locale files modified (Lokalise handles others)
- [x] RTL layout verified

### Accessibility (if UI changes)

- [x] Semantic HTML elements used
- [x] ARIA attributes added where needed
- [x] Keyboard navigation works

## Screenshots/Videos

| Before                                        | After                                                    |
| --------------------------------------------- | -------------------------------------------------------- |
|Dark mode | image |
|light mode | image |
|sepia mode | image |
|without answer  | image |
|mobile view | image |

## Related PRs

- Related to: `QF-3120-end-surah-readmore-card` (base branch)

## AI Assistance Disclosure

- [ ] AI tools were NOT used for this PR
- [x] AI tools were used, and I have **thoroughly reviewed and validated** all generated code

[QF-3120]: https://quranfoundation.atlassian.net/browse/QF-3120?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/QuranReader/EndOfSurahSection/ExploreCard/actions.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    icon: ChatIcon,
+    namespace: 'common',
+    modalType: ModalType.REFLECTION,
+    getNavigationUrl: ({ verseKey }) => getVerseReflectionNavigationUrl(verseKey),
+  },
+  {
+    key: 'lessons',
+    icon: LearningPlanIcon,
+    namespace: 'common',
+    modalType: ModalType.REFLECTION,
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply src/components/QuranReader/EndOfSurahSection/ExploreCard/actions.ts]: the reflections and lessons share the same model 
so the current configuration is correct

File: src/components/QuranReader/EndOfSurahSection/ExploreCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+  const getButtonLabel = (button: ActionButton): string => {
+    const namespace = button.namespace ?? 'quran-reader';
+    return t(`${namespace}:${button.key}`);
+  };
+
+  const handleButtonClick = useCallback(
+    (button: ActionButton) => {
+      onModalOpen(button.modalType);
+      logButtonClick(`end_of_surah_${button.key}`, { chapterNumber });
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply src/components/QuranReader/EndOfSurahSection/ExploreCard/index.tsx]: good catch

File: src/components/QuranReader/EndOfSurahSection/ExploreCard/actions.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  chapterNumber,
+  verseKey,
+  selectedTafsirs,
+}: {
+  chapterNumber: number;
+  verseKey: string;
+  selectedTafsirs: string[];
+}): string => {
+  const [, verseNumber] = getVerseAndChapterNumbersFromKey(verseKey);
+  const tafsirId = selectedTafsirs?.[0] as string;
```
[the remark this answers is not in the document]
Comment (yousefdergham) [reply src/components/QuranReader/EndOfSurahSection/ExploreCard/actions.ts]: the redux state for tasir always has at least one tafsir 
and there is also a default value 

so no need for this

**Comment by AhmedCodeGuy** [comment]
### [Correctness] `hasQuestions` check uses first verse only

**File:** `src/components/QuranReader/EndOfSurahSection/index.tsx:32`

```tsx
const verseKey = `${chapterNumber}:1`;
const hasQuestions = questionsData?.[verseKey]?.total > 0;
```

**Issue:** This only checks if verse 1 has questions, but the "End of Surah" section is shown after reading the entire chapter. Users might want to explore Q&A for any verse in the chapter they just finished.

**Impact:** The Answers button may be hidden even when questions exist for other verses in the chapter.

**Suggestion:** Consider checking if any verse in the chapter has questions, or aggregate question counts at the chapter level.

**Review by AhmedCodeGuy — APPROVED** [review]

**Review by AhmedCodeGuy — COMMENTED** [review]
#### 1. [Testing] Missing unit tests for ExploreCard component

The PR checklist marks "Unit tests added/updated" as complete, but no test files were found for the new `ExploreCard` component.

**Recommended test coverage:**
- ExploreCard renders with all action buttons
- "Answers" button is hidden when `hasQuestions` is false
- "Answers" button is visible when `hasQuestions` is true
- Click handlers trigger correct modal types
- Random suggestion displays when available
- "My Quran" link navigates to profile
